apps/pkcs12.c

```diff
@@ -541,13 +541,15 @@ int pkcs12_main(int argc, char **argv)
             X509_STORE_free(store);
 
             if (vret == X509_V_OK) {
+                int add_certs;
                 /* Remove from chain2 the first (end entity) certificate */
                 X509_free(sk_X509_shift(chain2));
                 /* Add the remaining certs (except for duplicates) */
-                if (!X509_add_certs(certs, chain2, X509_ADD_FLAG_UP_REF
-                                    | X509_ADD_FLAG_NO_DUP))
-                    goto export_end;
+                add_certs = X509_add_certs(certs, chain2, X509_ADD_FLAG_UP_REF
```

Comment:
Should it be actually `ret = ...` ?

Reply:
No. ret is already 1 (because this is an app).. the goto will leave this alone. If it all passes then it is set to zero as the last thing..
(sidenote: This is a stupidly long function).

Reply:
Ah, I was confused by the inverted meaning - that the ret is actually exit code.